### PR TITLE
Adds black check to travis-ci setup

### PR DESCRIPTION
**Pull request by JamesMura**
Adds a step to the travis-ci config to check that the code is formatted with [black](https://github.com/ambv/black)

This also fixes the formatting issues in the library with black

## Notes
Black requires python >=3.6 to run , but you can format code >= python 2

We should consider adding a pre commit hook or I am looking at what documentation I can update.

**_This looks like a big change but all of it is either ci config or formatting fixes_.**

This increases the line length limit for flake8 to 100 and disables the rule W503 so that flake8 and black do not clash.
## Related Issue
Add Black auto formatter #446

## Motivation and Context
This makes sure that the code is formatted in a standard way

## How Has This Been Tested?
There should be no changes to functionality , just formatting so the existing tests are sufficient

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rossmechanic** [comment]
Thanks for this @JamesMura . Do we want to include instructions for running this before push in a few places? I.e. the contributing file and the PR template?

File: AUTHORS.rst [hunk 1]
```diff
@@ -66,6 +66,7 @@ Authors
 - Adnan Umer (@uadnan)
 - Jonathan Zvesper (@zvesp)
 - Matheus Cansian (@mscansian)
+- James Muranga (@jamesmura)
```
Comment (barm) [inline AUTHORS.rst]: 🎉

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -28,7 +28,7 @@
 ## Checklist:
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
-- [ ] My code follows the code style of this project.
+- [ ] My code is auto formatted using black
```
Comment (barm) [inline PULL_REQUEST_TEMPLATE.md]: I'd prefer we just remove the checkbox because this is now enforced in CI

File: tox.ini [hunk 3]
```diff
@@ -8,8 +8,9 @@ envlist =
 
 
 [flake8]
-ignore = N802,F401
+ignore = N802,F401,W503
 max-complexity = 10
+max-line-length = 100
```
Comment (barm) [inline tox.ini]: We might want to make this 88 to match Black

**Comment by rossmechanic** [comment]
@barm @JamesMura Not sure why the patch didn't hit the coverage goal, but happy to merge this with project hitting coverage goal (only thing that actually needs to pass for us to merge is CI). @barm want to take a final look through this and then we can merge? Excited to have this feature!

**Comment by rossmechanic** [comment]
@JamesMura want to fix the conflicts? Otherwise I'm happy to take this over. Want to get this merged ASAP.

**Comment by JamesMura** [comment]
I won’t be able to get to this till later in the day

**Comment by rossmechanic** [comment]
@JamesMura no worries that's fine! Thanks

File: simple_history/management/commands/populate_history.py [hunk 4]
```diff
@@ -24,38 +26,37 @@ class Command(BaseCommand):
...
-            '--auto',
-            action='store_true',
-            dest='auto',
+            "--auto",
+            action="store_true",
+            dest="auto",
             default=False,
-            help='Automatically search for models with the '
-                 'HistoricalRecords field type',
+            help="Automatically search for models with the "
```
Comment (rossmechanic) [inline simple_history/management/commands/populate_history.py]: Don't love this reformatting here. Can we change?

File: simple_history/management/commands/populate_history.py [hunk 5]
```diff
@@ -24,38 +26,37 @@ class Command(BaseCommand):
...
-            action='store',
-            dest='batchsize',
+            "--batchsize",
+            action="store",
+            dest="batchsize",
             default=200,
             type=int,
-            help='Set a custom batch size when bulk inserting historical '
-                 'records.'
+            help="Set a custom batch size when bulk inserting historical " "records.",
```
Comment (rossmechanic) [inline simple_history/management/commands/populate_history.py]: Combine this string

File: simple_history/manager.py [hunk 6]
```diff
@@ -39,8 +39,10 @@ def most_recent(self):
         Returns the most recent copy of the instance available in the history.
         """
         if not self.instance:
-            raise TypeError("Can't use most_recent() without a %s instance." %
-                            self.model._meta.object_name)
+            raise TypeError(
+                "Can't use most_recent() without a %s instance."
+                % self.model._meta.object_name
```
Comment (rossmechanic) [inline simple_history/manager.py]: Want to change this to a `.format`? don't love the `%` on second line

**Comment by rossmechanic** [comment]
Hey @JamesMura do you still have time to work on this? Or want me to take it over?

**Comment by rossmechanic** [comment]
Closed by #480

**Pull request closed without merging**